src/main/java/io/jenkins/plugins/ml/Server.java

```diff
@@ -74,26 +83,78 @@ public long getMaxResults() {
         public String getDisplayName() {
             return "Server";
         }
+
+        public FormValidation doCheckServerName(@QueryParameter String serverName) {
+            if( Util.fixEmptyAndTrim(serverName) == null){
+                return FormValidation.warning("* Required ");
+            }
+            String regex = "^[a-zA-Z0-9_]+$";
+            Pattern pattern = Pattern.compile(regex);
```

Comment:
@Loghijiaha I think we can move the regex pattern compilation out of the function, as a constant in the class. That way we don't need to compile it with every request.

Reply:
Yes, That makes sense!